sycl/doc/extensions/proposed/sycl_ext_oneapi_graph.asciidoc

```diff
@@ -562,7 +562,10 @@ Exceptions:
 using namespace ext::oneapi::experimental;
 node add_free(void *data, const std::vector<node>& dep = {});
 ----
-|Adding a node that encapsulates a memory free operation.
+|Adds a node that encapsulates a memory free operation. The memory pointed to
+by `data` must be a valid address that has been allocated using one of the USM
```

Comment:
The phrase

> must be a valid address that has been allocated using one of the USM allocation routines

means that a user could call `add_free` using an address that was from a USM allocation function that is not `add_malloc_device`, but `sycl::malloc_[device,host,shared]`. I'd like to double check if this is behaviour we want. If not, this language can be changed to "must be a valid address that has been allocated from add_malloc_device"

Reply:
I think letting the graph free an allocation it didn't create is very error prone. IMO we should either mandate that it must have been allocated as part of the graph and error if not, or remove allocation/freeing altogether from graphs.